.github/workflows/build-pull-request.yml

```diff
@@ -20,29 +20,119 @@ defaults:
     shell: bash
 
 jobs:
-  build:
+  build-and-test-jvm:
     name: Build on ${{ matrix.os }} - ${{ matrix.java }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        java: [17, 21]
+        java: [17]
     runs-on: ${{ matrix.os }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
     steps:
       - name: Prepare git
         run: git config --global core.autocrlf false
         if: startsWith(matrix.os, 'windows')
 
       - uses: actions/checkout@v3
+
       - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
           java-version: ${{ matrix.java }}
           cache: 'maven'
 
       - name: Build with Maven
-        run: mvn -B clean install -Dno-format
+        run: ./mvnw -B clean install -Dno-format -ntp
+
+      - name: Zip the Maven repo
+        run: |
+          tar -czf ${{ runner.temp }}/maven-repo.tgz -C ~ .m2/repository
+          # Avoid caching our own artifacts
+          rm -Rf ~/.m2/repository/io/quarkiverse/langchain4j
+      - name: Persist the Maven repo
+        uses: actions/upload-artifact@v4
+        with:
+          name: maven-repo
+          path: ${{ runner.temp }}/maven-repo.tgz
+          retention-days: 1
```

Comment:
I would advise to keep them for 5 days. I used to be very aggressive in Quarkus and then it became very unpractical when you want to rerun a specific subjob after a week-end.

Reply:
+1, increased to 5 days